src/ansible_navigator/actions/builder.py

```diff
@@ -0,0 +1,95 @@
+"""Run the builder subcommand."""
+import os
+import shutil
+
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from ..app import App
+from ..configuration_subsystem import ApplicationConfiguration
+from ..configuration_subsystem.definitions import Constants
+from ..runner import Command
+from . import _actions as actions
+
+
+@actions.register
+class Action(App):
+    """Run the builder subcommand."""
+
+    KEGEX = "^b(?:uilder)?$"
+
+    def __init__(self, args: ApplicationConfiguration):
+        """Initialize the action.
+
+        :param args: The current application configuration.
+        """
+        super().__init__(args=args, logger_name=__name__, name="builder")
+        self._runner: Command
+
+    def run_stdout(self) -> Union[None, int]:
+        """Run in mode stdout.
+
+        :returns: The return code or None
+        """
+        self._logger.debug("builder requested in stdout mode")
+        response = self._run_runner()
+        if response:
+            _, _, ret_code = response
+            return ret_code
+        return None
+
+    def _run_runner(self) -> Optional[Tuple]:
+        """Spin up runner.
+
+        :return: The stdout, stderr and return code from runner
+        """
+        if isinstance(self._args.set_environment_variable, dict):
+            envvars_to_set = self._args.set_environment_variable.copy()
+        elif isinstance(self._args.set_environment_variable, Constants):
+            envvars_to_set = {}
+        else:
+            log_message = (
+                "The setting 'set_environment_variable' was neither a dictionary"
+                " or Constants, please raise an issue. No environment variables will be set."
+            )
+            self._logger.error(
+                "%s The current value was found to be '%s'",
+                log_message,
+                self._args.set_environment_variable,
+            )
+        envvars_to_set = {}
+
+        if self._args.display_color is False:
+            envvars_to_set["ANSIBLE_NOCOLOR"] = "1"
+
+        if self._args.execution_environment:
+            self._logger.info("For builder subcommand execution-environment is disabled")
+
+        kwargs = {
+            "execution_environment": False,
+            "host_cwd": os.path.abspath(os.path.expanduser(self._args.workdir)),
+            "navigator_mode": self._args.mode,
+            "pass_environment_variable": self._args.pass_environment_variable,
+            "set_environment_variable": envvars_to_set,
+            "timeout": self._args.ansible_runner_timeout,
+        }
+
+        ansible_builder_path = shutil.which("ansible-builder")
```

Comment:
(optional) `"ansible-builder"` is used in multiple places so I'd probably suggest putting it in a class attribute.
Also, it seems like it's best to do the resolution in the initializer and expose the full path as an instance attribute. The `is None` check would need to remain in this method, though.

Reply:
`ansible-builder` executable path is used only in `_run_runner` method and the path information is required only while invoking `ansible-runner` API hence it is not a class attribute.

Reply:
Changing it later if needed shouldn't be a major issue. I think Ganesh's comment about scope here is accurate. If we need it commonly between multiple funcs later it should be a quick change.